modules/bloxone_infra_host_gcp/variables.tf

```diff
@@ -0,0 +1,101 @@
+variable "join_token" {
+  description = "The join token to use for the BloxOne Host. If not provided, a join token will be created."
+  type        = string
+  default     = null
+}
+
+variable "name" {
+  description = "The name of the virtual machine"
+  type        = string
+}
+
+variable "project" {
+  description = "The name of the GCP project"
+  type        = string
+
+}
+
+variable "region" {
+  description = "The region where the resources will be created"
+  type        = string
+}
+
+variable "zone" {
+  description = "The zone where the resources will be created"
+  type        = string
+}
+
+variable "machine_type" {
+  description = "The machine type to use for the virtual machine"
+  type        = string
+  default     = "e2-standard-4"
+}
+
+variable "network_interfaces" {
+  description = "List of network interfaces to be attached to the virtual machine."
+  type        = list(object({
+    network          = string
+    subnetwork       = string
+    assign_public_ip = optional(bool)
+  }))
+}
+
+variable "disk_type" {
+  description = "The type of the data disk."
+  type        = string
+  default     = "pd-standard"
+}
+
+variable "source_image" {
+  description = "The source image to use for the virtual machine."
+  type        = string
+}
```

Comment:
A default would be useful here

Reply:
added variable and default value ( with validation )